plugins/outputs/sensu/sensu.go

```diff
@@ -0,0 +1,500 @@
+## $ sensuctl user create telegraf --group telegraf --password REDACTED 
+## $ sensuctl api-key grant telegraf
+##
+## For more information on Sensu RBAC profiles & API tokens, please visit: 
+## - https://docs.sensu.io/sensu-go/latest/reference/rbac/
+## - https://docs.sensu.io/sensu-go/latest/reference/apikeys/ 
+## 
+# api_key = "${SENSU_API_KEY}"
+
+## Optional TLS Config
+# tls_ca = "/etc/telegraf/ca.pem"
+# tls_cert = "/etc/telegraf/cert.pem"
+# tls_key = "/etc/telegraf/key.pem"
+## Use TLS but skip chain & host verification
+# insecure_skip_verify = false
+
+## Timeout for HTTP message
+# timeout = "5s"
+
+## HTTP Content-Encoding for write request body, can be set to "gzip" to
+## compress body or "identity" to apply no encoding.
+# content_encoding = "identity"
+
+## Sensu Event details
+##
+## Below are the event details to be sent to Sensu.  The main portions of the
+## event are the check, entity, and metrics specifications. For more information
+## on Sensu events and its components, please visit:
+## - Events - https://docs.sensu.io/sensu-go/latest/reference/events
+## - Checks -  https://docs.sensu.io/sensu-go/latest/reference/checks
+## - Entities - https://docs.sensu.io/sensu-go/latest/reference/entities
+## - Metrics - https://docs.sensu.io/sensu-go/latest/reference/events#metrics
+##
+## Check specification
+## The check name is the name to give the Sensu check associated with the event
+## created.
+[outputs.sensu-go.check]
+  name = "telegraf"
+
+## Entity specification
+## Configure the entity name and namepsace, if necessary.
+##
+## NOTE: if the output plugin is configured to send events to a
+## backend_api_url and entity_name is not set, the value returned by
+## os.Hostname() will be used; if the output plugin is configured to send
+## events to an agent_api_url, entity_name and entity_namespace are not used.
+# [outputs.sensu-go.entity]
+#   name = "server-01"
+#   namespace = "default"
+
+## Metrics specification
+## Configure the tags for the metrics that are sent as part of the Sensu event
+# [outputs.sensu-go.tags]
+#   source = "telegraf"
+
+## Configure the handler(s) for processing the provided metrics
+# [outputs.sensu-go.metrics]
+#   handlers = ["influxdb","elasticsearch"]
+`
+
+// Description provides a description of the plugin
+func (s *Sensu) Description() string {
+	return "Send aggregate metrics to Sensu Monitor"
+}
+
+// SampleConfig provides a sample configuration for the plugin
+func (s *Sensu) SampleConfig() string {
+	return sampleConfig
+}
+
+func (s *Sensu) createClient(ctx context.Context) (*http.Client, error) {
+	tlsCfg, err := s.ClientConfig.TLSConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	client := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: tlsCfg,
+		},
+		Timeout: s.Timeout.Duration,
+	}
+
+	return client, nil
+}
+
+func (s *Sensu) Connect() error {
+	if s.Timeout.Duration == 0 {
```

Comment:
I guess zero is not a valid value here (e.g. wait forever)? If however, zero is a valid value, then please set the default value in `init()` when constructing the struct. Asking, just in case... :-)

Reply:
Yeah, this seems redundant and zero should be a valid value.  I think I'll remove this portion and leave the setting of Timeout in `init()`.

Reply:
Sounds reasonable.